src/main/java/org/jabref/logic/integrity/PredatoryJournalChecker.java

```diff
@@ -0,0 +1,54 @@
+package org.jabref.logic.integrity;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.jabref.logic.journals.PredatoryJournalRepository;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+
+public class PredatoryJournalChecker implements EntryChecker {
+
+    private final List<Field> fieldsToCheck;
+    private final PredatoryJournalRepository predatoryJournalRepository;
+
+    public PredatoryJournalChecker(PredatoryJournalRepository predatoryJournalRepository, Field... fieldsToCheck) {
```

Comment:
JabRef's "style" is to pass Lists. With `List.of(...)` this should be "easy" for the caller.

Then you can also remove the for loop from lines 24 to 27